src/puppetlabs/puppetdb/query/population.clj

```diff
@@ -13,10 +13,9 @@
 ;;; Query library
 
 (def ^:private from-all-resources-and-their-nodes
-  (hcore/build :from [:catalogs :catalog_resources :certnames]
-               :merge-where [:and
-                             [:= :catalogs.id :catalog_resources.catalog_id]
-                             [:= :certnames.certname :catalogs.certname]]))
+  (hcore/build :from [:certnames]
+               :join [:catalogs [:= :certnames.certname :catalogs.certname]
```

Comment:
We're filing this under tech debt and addressing it later, reports/edges/resources_events/catalogs/catalog_resources, are all inconsistent.